libpromises/cf3lex.l

```diff
@@ -252,11 +258,11 @@ promise_type   [a-zA-Z_]+:
                                    return 0;
                                  }
                                  P.if_depth--;
-                               }
-
-<if_ignore_state>[^@]      {
-                               ParserDebug("\tL:inside macro @if, ignoring text:%s\n", yytext);
-                           }
+                               }\
+<if_ignore_state>.    {
+                          /* eat up al unknown chars when line starts with @*/
+                          ParserDebug("\tL:inside macro @if, ignoring text:%s\n", yytext);
+                      }
```

Comment:
This is one reads the whole line instead per char but skips line starting with @ then it switches to per char read and parse all chars and seea the @endiff. When I turned on the Lex debugger I was surprised by the one char parsing when in ignore mode. Was difficult to  follow 😀